File: apps/backend/src/EnterpriseAgentOs.Domain/Agents/IAgentTemplateService.cs
namespace EnterpriseAgentOs.Domain.Agents;

public interface IAgentTemplateService
{
    Task<IReadOnlyList<AgentTemplateDto>> ListAsync(CancellationToken ct = default);
    Task<AgentTemplateDto?> GetAsync(Guid id, CancellationToken ct = default);
    Task<AgentDto> CreateAgentFromTemplateAsync(
        Guid templateId,
        string name,
        string provider,
        string? model,
        Guid ownerId,
        CancellationToken ct = default);
}

File: apps/backend-java/src/main/java/com/enterpriseagentos/backendjava/domain/features/agents/IAgentTemplateService.java
package com.enterpriseagentos.backendjava.domain.features.agents;

import java.util.List;
import java.util.UUID;
import java.util.concurrent.CompletableFuture;

public interface IAgentTemplateService {
    CompletableFuture<List<AgentTemplateDto>> listAsync();
    CompletableFuture<AgentTemplateDto> getAsync(UUID id);
}
